docs/src/main/tut/poly.md

```diff
@@ -0,0 +1,229 @@
+  
+  
+# An example: lists
+  
+```tut
+import cats.Functor
+import cats.data.Types._
+import cats.data._
+import cats.fix.Types._
+import cats.fix.Fix
+
+```
+
+Let's start with some convenient *type aliases* (whose names you'll probably recognize)
+  
+```tut
+  type Nil[Y, Z] = Constant[Unit, Z]
+  type Head[Y, Z] = Constant[Y, Z]
+  type Tail[Y, Z] = Identity[Y, Z]
+  type Cons[Y, Z] = Prod[λ[ζ => Head[Y, ζ]], λ[ζ => Tail[Y, ζ]], Z]
+  type List[Y, Z] = Xor[Nil[Y, Z], Cons[Y, Z]]
+
+```
+ 
+Next we can define corresponding *functors*
+ 
+```tut
+  def nilFunctor[Y]: Functor[λ[ζ => Nil[Y, ζ]]] = new ConstantFunctor[Unit]
+  def headFunctor[Y]: Functor[λ[ζ => Head[Y, ζ]]] = new ConstantFunctor[Y]
+  def tailFunctor[Y]: Functor[λ[ζ => Tail[Y, ζ]]] = new IdentityFunctor[Y]
+  def consFunctor[Y]: Functor[λ[ζ => Cons[Y, ζ]]] = new ProductFunctor[Head, Tail, Y](headFunctor[Y], tailFunctor[Y])
+  
+```
+
+and an implicit functor
+
+```tut
+  implicit def listFunctor[Y]: Functor[λ[ζ => List[Y, ζ]]] = new SumFunctor[Nil, Cons, Y](nilFunctor[Y], consFunctor[Y])
+  
+```
+
+now we are ready to define corresponding *constructors*
+
+```tut
+  def cons[Y, Z](y: Y, z: Z): Cons[Y, Z] = Prod[λ[ζ => Head[Y, ζ]], λ[ζ => Tail[Y, ζ]], Z](y, z)
+
+  def nilList[Y, Z]: Unit => List[Y, Z] = _ => Xor.Left[Nil[Y, Z]](())
+  def consList[Y, Z]: (Y, Z) => List[Y, Z] = (y, z) => Xor.Right[Cons[Y, Z]](cons(y, z))
+
+```
+
+and, once we have defined the *least fixed point*
+
+```tut
+  type FixList[Z] = Fix[λ[ζ => List[Z, ζ]]]
+
+```
+
+and corresponding *function*
+```tut
+  def fixList[Z]: List[Z, FixList[Z]] => FixList[Z] = Fix[λ[ζ => List[Z, ζ]]]
+  
+```
+
+we can define corresponding constructors
+
+```tut
+  def nil[Y](implicit implicitListFunctor: Functor[λ[ζ => List[Y, ζ]]]): Unit => FixList[Y] =
+    _ => fixList[Y](nilList[Y, FixList[Y]](()))
+  def cons[Y](implicit implicitListFunctor: Functor[λ[ζ => List[Y, ζ]]]): (Y, FixList[Y]) => FixList[Y] =
+    (y, ys) => fixList[Y](consList[Y, FixList[Y]](y, ys))
+    
+```
+
+Finally we are ready for some *algebra*'s
+```tut
+  type ListAlgebra[Y, Z] = Algebra[λ[ζ => List[Y, ζ]], Z]
+
+  def listAlgebra[Y, A]: (A, (Y, A) => A) => List[Y, A] => A = (a, ya2a) =>
+    _.fold(nil => a, cons => ya2a(cons.first, cons.second))
+
+  def showListAlgebra[Y]: ListAlgebra[Y, String] =
+    listAlgebra("Nil", _ + " :: " + _)
+ 
+  def incrIntListAlgebra: ListAlgebra[Int, FixList[Int]] =
+    listAlgebra(nil[Int].apply(()), (i, is) => cons[Int].apply(i + 1, is))  
+     
+```
+
+and some corresponding *structurally recursive functions on lists*
+```tut
+  def showFixList[Y]: FixList[Y] => String =
+    _.cata[String](showListAlgebra)
+
+  def incrIntFixList: FixList[Int] => FixList[Int] =
+    _.cata[ FixList[Int]](incrIntListAlgebra)
+
+```
+
+and we can use all we defined as follows:
+evaluating `showFixList[Int](incrIntFixList(cons[Int].apply(1, cons[Int].apply(2, nil[Int].apply(())))))` 
```

Comment:
Hello, the `apply` is necessary because of the implicit functor argument (a somewhat unfortunate thing in Scala).
Maybe there is a way around this, I'll have a look